### PR TITLE
Improve ESlint structure

### PR DESCRIPTION
**Pull request by luizbaldi**
Lint all files according to Airbnb style guide, I also added `lint-staged` and `husky` with a simple _pre-commit_ hook running ESLint.

The goal of this PR is to have a more standardized codebase :)

### CODE REVIEW

File: src/components/List/List.js [hunk 1]
```diff
@@ -46,39 +41,44 @@ const List = ({
...
 
 List.defaultProps = {
   style: {},
   fullWidth: false,
   shadow: true,
-  inline: false
+  inline: false,
+  className: '',
+  children: null,
+  verticalAlign: 'bottom',
```
Comment (arturbien) [inline src/components/List/List.js]: Let's remove default values for `verticalAlign` and `horizontalAlign` mate 🎉 it was left out on purpose

File: src/components/List/List.js [hunk 2]
```diff
@@ -1,39 +1,34 @@
...
+import React from 'react';
+import propTypes from 'prop-types';
 
-import styled from "styled-components";
-import { createBorderStyles, createBoxStyles } from "../common";
+import styled from 'styled-components';
+import { createBorderStyles, createBoxStyles } from '../common';
 
-const createListPositionStyles = ({
-  verticalAlign = "bottom",
```
Comment (arturbien) [inline src/components/List/List.js]: and lets keep the default values in `createListPositionStyles` (it's made that way in order to make component position absolute only when user specifies either one of: `verticalAlign` and `horizontalAlign`). We'll change API for that component in V3

File: package.json [hunk 3]
```diff
@@ -55,16 +56,20 @@
     "@testing-library/jest-dom": "^4.0.0",
     "@testing-library/react": "^8.0.4",
     "@types/jest": "^24.0.15",
+    "babel-eslint": "^10.0.2",
     "babel-loader": "^8.0.5",
     "babel-plugin-styled-components": "^1.10.0",
+    "classnames": "^2.2.6",
```
[the remark this answers is not in the document]
Comment (luizbaldi) [reply package.json]: of course! 

fixed.

File: .eslintrc.js [hunk 4]
```diff
@@ -1,7 +1,23 @@
...
+  },
+  rules: {
+    "import/no-extraneous-dependencies": ["error", { devDependencies: true }],
+    "react/jsx-filename-extension": [1, { extensions: [".js", ".jsx"] }],
+    "jsx-a11y/label-has-associated-control": [ "error", {
+      "required": {
+        "some": [ "nesting", "id"  ]
+      }
+    }],
+    "jsx-a11y/label-has-for": [ "error", {
```
[the remark this answers is not in the document]
Comment (luizbaldi) [reply .eslintrc.js]: awesome!

`label-has-for` is used on Airbnb style-guide, I disabled that rule and added `label-has-associated-control` :)

File: src/components/List/List.js (hunk 1 above)
Comment (luizbaldi) [reply src/components/List/List.js]: ok, fixed!

File: src/components/List/List.js (hunk 2 above)
Comment (luizbaldi) [reply src/components/List/List.js]: sure, gonna add a temporary `undefined` value to those properties so we don't have to disable the ESlint rule for props without default values and we fix that on v3 :)

**Pull request merged**